bigframes/bigquery/__init__.py

```diff
@@ -239,6 +239,15 @@ def json_extract(
     return series._apply_unary_op(ops.JSONExtract(json_path=json_path))
 
 
+def struct(value: bigframes.dataframe.DataFrame) -> series.Series:
+    data: List[Dict[str, Any]] = [{} for _ in value.index]
```

Comment:
it creates a local data copy in memory. It won't fit if the original DF is a large table in BQ. @chelsea-lin Do you have any suggestions?

Reply:
just so I can learn: does struct.explode() [code pointer: https://github.com/googleapis/python-bigquery-dataframes/blob/main/bigframes/operations/structs.py#L43] create a local data copy in memory? @GarrettWu cc @tswast @TrevorBergeron as Tim introduced this change [PR: https://github.com/googleapis/python-bigquery-dataframes/pull/71] and Trevor reviewed

Reply:
If `explode()` does create a local copy, I think we are okay to create local copy for `struct`, as the use case of this function is to unexplode a Dataframe back into a Series of structs (see b/357588049).

An alternative idea was to use `DataFrame.loc` but seems like that has some unnecessary overhead and also creates local copies through calling `to_pandas()` when Series is a BigFrames Series

Reply:
The majority of BigFrames operators are designed to be deferred. This means that we construct static expression tree representing your operations, and these trees won't be executed (causing compiler and data downloads) until you explicitly trigger it, typically within actions like `to_pandas`. Taken `struct.explode()` as example, it calls `struct.field` and `series.rename`, adhere to this deferred model as well. You can confirm this behavior by https://screenshot.googleplex.com/Pqf69h7ZKUTmHGP, where no BQ job for `s.struct.explode()`.

Following Gerrett's suggestion, we can implement this operator by creating a new STRUCT operator. A similar implementation approach can be seen in [this pull request](https://github.com/googleapis/python-bigquery-dataframes/pull/868/files), where a JSONExtract unary operation is defined. This operation takes a single argument, json_path, and its compiler rule (defined in the `json_extract` method) generates the SQL expression `JSON_EXTRACT(json_obj, json_path)`.